CodingTracker.TildaDares/UserInterface/MainMenu.cs

```diff
@@ -0,0 +1,324 @@
+    {
+        Console.Clear();
+        if (!HasCodingSessions())
+        {
+            Input.ContinueMenu();
+            return;
+        }
+    
+        GetCodingSessions();
+        var id = AnsiConsole.Ask<int>("Enter the coding session ID you wish to delete:");
+        var confirmation = Input.ConfirmPrompt("[yellow]This action is irreversible. Confirm delete?[/]");
+        if (!confirmation) return;
+        
+        _database.DeleteCodingSession(id);
+        Input.ContinueMenu();
+    }
+
+    private void StartCodingSession()
+    {
+        Console.Clear();
+        var stopwatchService = new StopwatchService();
+        AnsiConsole.MarkupLine("[green]Press any key to stop the coding session...[/]");
+        stopwatchService.Start();
+        Console.ReadKey();
+        stopwatchService.Stop();
+        AnsiConsole.MarkupLine($"[green]You ran the coding session for: {stopwatchService.Duration}...[/]");
+        
+        var confirmation = Input.ConfirmPrompt("[yellow]Save coding session to database?[/]");
+        if (!confirmation)
+        {
+            Input.ContinueMenu();
+            return;
+        }
+
+        Console.Clear();
+        var codingSession = new CodingSession(stopwatchService.StartTime, stopwatchService.EndTime);
+        _database.InsertCodingSession(codingSession);
+        Input.ContinueMenu();
+    }
+
+    private void FilterCodingSessionsByPeriod()
+    {
+        Console.Clear();
+        if (!HasCodingSessions())
+        {
+            Input.ContinueMenu();
+            return;
+        }
+        
+        var startTime = Input.GetDateInput(
+            "[green]Enter the start date & time of sessions you want to filter in the format[/] [blue]dd/mm/yyyy HH:mm (24-hour format only)[/]:\n");
+        Console.Clear();
+        
+        var endTime = Input.GetDateInput(
+            "[green]Enter the end date & time of sessions you want to filter in the format[/] [blue]dd/mm/yyyy HH:mm (24-hour format only)[/]:\n", minRange: startTime);
+        Console.Clear();
+
+        var filter = new CodingSessionFilter{ StartTime = startTime, EndTime = endTime };
+        var sessions = _database.GetAllCodingSessions(filter);
+        
+        if (sessions.Count == 0)
+        {
+            AnsiConsole.MarkupLine("[green]No coding sessions found![/]");
+            Input.ContinueMenu();
+            return;
+        }
+        
+        DisplayCodingSessions(sessions,$"All coding sessions in the period {startTime} to {endTime}:");
+        Input.ContinueMenu();
+    }
+
+    private void ViewCodingSessionReportByPeriod()
+    {
+        Console.Clear();
+        if (!HasCodingSessions())
+        {
+            Input.ContinueMenu();
+            return;
+        }
+        
+        var startTime = Input.GetDateInput(
+            "[green]Enter the start date & time of sessions you want to view a report for in the format[/] [blue]dd/mm/yyyy HH:mm (24-hour format only)[/]:\n");
```

Comment:
🟠 Note that you have a configurable dateFormat in your `app.config` but a hard-coded message text here. A user could localise the app.config e.g. `yyyy-MM-dd HH:mm`, which would then make this message false and mis-leading.

🔧 Do what you have done in `Input.cs`, read the `app.config` value and user string interpolation to show the variable format rather than hard-coded format.